src/value.zig

```diff
@@ -1490,7 +1509,6 @@ pub const Value = extern union {
                     const field_index = ty.unionTagFieldIndex(val.unionTag(), mod);
                     const field_type = ty.unionFields().values()[field_index.?].ty;
                     const field_val = val.fieldValue(field_type, field_index.?);
-
                     return field_val.writeToPackedMemory(field_type, mod, buffer, bit_offset);
                 },
             },
```

Comment:
Assuming what I did in this PR isn't completely wrong (I think some of it is), what else would be missing here now? Could we make the `else => @panic("TODO implement writeToPackedMemory for more types"),` below an `else => unreachable,` and get rid of `error.Unimplemented` and replace it with `unreachable` as well?